src/commands/create/index.ts

```diff
@@ -65,130 +93,170 @@ export default class Create extends Command {
-                    }
-                },
-                type: 'input',
+                name: 'extensionParameter',
+                prefix: '>',
+                message: parameter.description,
+                default: parameter.default,
+                type: parameter.type,
             };
         },
     };
 
-    appManifestInterfaces: AppManifestInterfaces = { interfaces: [] };
+    private async _configureCore(availableCores: Core[]): Promise<CorePromptResult> {
+        let corePromptResult: CorePromptResult = {
+            chosenCore: undefined,
+            example: undefined,
+            appName: undefined,
+        };
+        let promptResult = await inquirer.prompt(Create.prompts.core(availableCores));
+
+        corePromptResult.chosenCore = promptResult.core as Core;
+
+        const sets = corePromptResult.chosenCore.parameterSets;
+        if (sets !== undefined) {
+            let chosenParamSetId = 0;
+            if (sets.length > 0) {
+                const parameterSetPromptResult = await inquirer.prompt(Create.prompts.parameterSets(sets));
+                chosenParamSetId = parameterSetPromptResult.parameterSet;
+            }
 
-    private _filterAvailableExamplesByLanguage(language: string) {
-        const filteredExamples = availableExamples.filter((example) => example.language === language);
-        if (!filteredExamples.length) {
-            throw new Error(`No example for your chosen language '${language}' available`);
+            for (const parameter of sets[chosenParamSetId].parameters) {
+                const coreParametersPromptResult = await inquirer.prompt(Create.prompts.coreParameters(parameter));
+                if (parameter.id === 'example') {
+                    corePromptResult.example = true;
+                    corePromptResult.appName = coreParametersPromptResult.coreParameter;
+                }
+                if (parameter.id === 'name') {
+                    corePromptResult.example = false;
+                    corePromptResult.appName = coreParametersPromptResult.coreParameter;
+                }
+            }
         }
-        return filteredExamples;
+        return corePromptResult;
     }
 
-    private async _runInteractiveMode(flags: any) {
-        const interactiveResponses: any = await inquirer.prompt([Create.prompts.language, Create.prompts.exampleQuestion]);
+    private async _configureExtension(
+        packageIndex: PackageIndex,
+        corePromptResult: CorePromptResult,
+    ): Promise<AppManifestInterfaceEntry[]> {
+        const appManifestInterfaceEntries: AppManifestInterfaceEntry[] = [];
+        if (corePromptResult.example) {
+            return appManifestInterfaceEntries;
+        }
+        const availableExtensions = packageIndex
+            .getExtensions()
+            .filter((ext: Extension) =>
+                ext.compatibleCores.find((compatibleCore: string) => compatibleCore === corePromptResult.chosenCore!.id),
```

Comment:
To avoid misuse `chosenCore` should be non-optional / mandatory. `!` should not be necessary here when configuring extensions for a definitely chosen core.